### PR TITLE
[App Portal] Fix users needing to verify their email for OAuth accounts

### PR DESCRIPTION
**Pull request by basokant**
- tested locally by deleting my account/user/session/application from the db, and signing in again. It no longer redirects me to /verify, skipping that flow entirely.

# Checklist

- [x] If any frontend changes were made, include a screenshot/demo in the overview
- [x] Checked all uses of changed component(s)/function(s)
- [x] If any changes were made to our backend services, at least one (happy path) unit test was added OR a `TODO` comment was added to create one.
- [x] Check that CI is passing.
- [x] If no (re-)reviews after 1-2 days, ping the web leads on the discord to remind them to review the PR.

@basokant @ArsalaanAli

### CODE REVIEW

**Review by hunterchen7 — CHANGES_REQUESTED** [review]
fix build, see comment

**Review by hunterchen7 — APPROVED** [review]
lgtm

File: src/utils/redirect.ts [hunk 1]
```diff
@@ -114,9 +122,13 @@ export const isVerifiedRedirect = async (
 
   const user = await db.query.users.findFirst({
     where: (users, { eq }) => eq(users.id, session?.user.id),
+    with: {
+      accounts: true,
+    },
```
Comment (basokant) [inline src/utils/redirect.ts:127]:
```suggestion
```

File: src/utils/redirect.ts [hunk 2]
```diff
@@ -82,9 +82,17 @@ export const notVerifiedRedirect = async (
...
```
Comment (basokant) [inline src/utils/redirect.ts:87]:
```suggestion
```

**Review by hunterchen7 — APPROVED** [review]
lgtm 🤑

**Pull request merged**